pallets/connectors-gateway/src/tests.rs

```diff
@@ -0,0 +1,480 @@
+use cfg_mocks::*;
+use cfg_traits::connectors::{Codec, OutboundQueue};
+use cfg_types::domain_address::*;
+use frame_support::{assert_noop, assert_ok};
+use sp_core::{crypto::AccountId32, ByteArray, H160};
+use sp_runtime::DispatchError::BadOrigin;
+
+use super::{
+	mock::{RuntimeEvent as MockEvent, *},
+	origin::*,
+	pallet::*,
+};
+
+mod utils {
+	use super::*;
+
+	pub fn get_random_test_account_id() -> AccountId32 {
+		rand::random::<[u8; 32]>().into()
+	}
+
+	pub fn event_exists<E: Into<MockEvent>>(e: E) {
+		let actual: Vec<MockEvent> = frame_system::Pallet::<Runtime>::events()
+			.iter()
+			.map(|e| e.event.clone())
+			.collect();
+
+		let e: MockEvent = e.into();
+		let mut exists = false;
+		for evt in actual {
+			if evt == e {
+				exists = true;
+				break;
+			}
+		}
+		assert!(exists);
+	}
+}
+
+use utils::*;
+
+mod set_domain_router {
+	use super::*;
+
+	#[test]
+	fn success() {
+		new_test_ext().execute_with(|| {
+			let domain = Domain::EVM(0);
+			let router = DomainRouterMock::new();
+
+			assert_ok!(ConnectorsGateway::set_domain_router(
+				RuntimeOrigin::root(),
+				domain.clone(),
+				router.clone(),
+			));
+
+			let storage_entry = DomainRouters::<Runtime>::get(domain.clone());
+			assert_eq!(storage_entry.unwrap(), router);
+
+			event_exists(Event::<Runtime>::DomainRouterSet { domain, router });
+		});
+	}
+
+	#[test]
+	fn bad_origin() {
+		new_test_ext().execute_with(|| {
+			let domain = Domain::EVM(0);
+			let router = DomainRouterMock::new();
+
+			assert_noop!(
+				ConnectorsGateway::set_domain_router(
+					RuntimeOrigin::signed(get_random_test_account_id()),
```

Comment:
I would keep the `AccountID` type for this mock runtime as `AccountID32` so that it easier to create an EVM address which is actually a truncated `AccountID32`.

Reply:
Sure! I wrote `Account` there more or less "randomly"